### ISSUE TITLE
Equivalent TypeVar cannot be assigned

### ISSUE THREAD

**Issue by tekumara** [microsoft/pyright#869]
```python
from typing import Generic, TypeVar

D = TypeVar('D', bool, int, float, object)
E = TypeVar('E', bool, int, float, object)

class Gen(Generic[D]):
    pass

GenAlias = Gen[E]
```

pyright 1.1.54 generates
```
9:15 - error: Type "E" cannot be assigned to TypeVar "D"
  Type "Unknown" is not compatible with constrained TypeVar "D" 
```

mypy is cool with this.

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next version of Pyright & Pylance.

**Comment by erictraut** [comment]
This is now implemented in Pyright 1.1.55, which I just published. It will also be included in the next release of Pylance.
